rollup/internal/controller/watcher/batch_proposer.go

```diff
@@ -279,14 +302,7 @@ func (p *BatchProposer) proposeBatch() error {
 				"maxUncompressedBatchBytesSize", p.maxUncompressedBatchBytesSize)
 
 			batch.Chunks = batch.Chunks[:len(batch.Chunks)-1]
-
-			metrics, err := utils.CalculateBatchMetrics(&batch, codecVersion)
-			if err != nil {
-				return fmt.Errorf("failed to calculate batch metrics: %w", err)
-			}
-
-			p.recordAllBatchMetrics(metrics)
-			return p.updateDBBatchInfo(&batch, codecVersion, *metrics)
+			return p.updateDBBatchInfo(&batch, codecVersion)
```

Comment:
This is acceptable, but in the happy path (when the compatibility check passes), we could just use `metrics` from here. In your current version, we need to run `CalculateBatchMetrics` 2x for each batch.